elastic/kibana
# [Search] Integrate "Send to background" UI with session service

## Summary

Part of https://github.com/elastic/kibana/issues/83640

This pr connects https://github.com/elastic/kibana/issues/81004 + https://github.com/elastic/kibana/pull/81099 making send to background indicator/popover more-or-less functional and we can see how the feature is shaping up.

![Screenshot 2020-11-19 at 16 51 55](https://user-images.githubusercontent.com/7784120/99690040-f720b700-2a87-11eb-9196-545edbc8deb0.png)


### Notes

* Integrates only with discover and dashboard 
* Only absolute time ranges 
* Background session management is not implemented yet, so link goes to nowhere 
* There are already some known/existing bugs and follow-up enhancements, I am keeping a list here https://github.com/elastic/kibana/issues/83640. 
  * A notable bug: on a dashboard on some state changes (e.g. switching view/edit mode) background session indicator disappears until refresh is hit. I want to fix it separately as it is more of a dashboard change, then search layer change.
* ~There is some TODOs in functional tests. Some units tests are missing. I will continue while it is being reviewed.~ **DONE** 
* Client side session service split into two services. One is pure CRUD mostly for management screen. Another is tracking state of current session. 

### How to test

```yml
xpack.data_enhanced.search.sendToBackground.enabled: true
data.search.aggs.shardDelay.enabled: true # helpful to create "slow" visualizations
```

To try a restore flow you can build a restore session URL, by getting sessionId from inspector and appending to the URL: `&searchSessionId=${your-id-from-inspector}`. (See tests)


### Checklist


### For maintainers
